### PR TITLE
feat: New `sqliteIsTransacting()` that returns if a transaction is active on the current connection

### PR DESCRIPTION
**Pull request by bpvgoncalves**
In response to issue #453 I implemented a functionality to allow checking of transaction status. 
I cannot find any develop branch so the PR targets the main branch.

It passes the tests:
```
══ Results ═════════════════════════════════════════════════════════════════════════════════════════════════════════
Duration: 35.5 s

── Skipped tests  ──────────────────────────────────────────────────────────────────────────────────────────────────
• astyle not found (1)
• No schemas available (1)
• NYI (1)
• On Linux (2)
• tweak: !date_typed (8)
• tweak: !time_typed (5)
• tweak: !timestamp_typed (8)

[ FAIL 0 | WARN 3 | SKIP 26 | PASS 7476 ]
```
It CMD CHECKs in my local machine (only 1 NOTE unrelated to the changes)
```
── R CMD check results ───────────────────────────────────────────────────────────────────── RSQLite 2.3.1.9002 ────
Duration: 2m 11s

❯ checking compilation flags used ... NOTE
  Compilation used the following non-portable flag(s):
    ‘-Wformat’ ‘-Wp,-D_FORTIFY_SOURCE=2’

0 errors ✔ | 0 warnings ✔ | 1 note ✖

R CMD check succeeded
```

Best Regards

### CODE REVIEW

File: R/transactions.R [hunk 1]
```diff
@@ -9,6 +9,8 @@ NULL
 #' auto-commit on.
 #' [DBI::dbWithTransaction()] is a convenient wrapper that makes sure that
 #' `dbCommit()` or `dbRollback()` is called.
+#' An helper function `sqliteIsTransacting()` is available to check the current
```
Comment (krlmlr) [inline R/transactions.R:12]:
```suggestion
#' A helper function `sqliteIsTransacting()` is available to check the current
```

File: src/connection.cpp [hunk 2]
```diff
@@ -67,6 +67,22 @@ void connection_release(cpp11::external_pointer<DbConnectionPtr> con_) {
   // don't release here to make sure a nice error message is delivered
 }
 
+[[cpp11::register]]
+bool connection_in_transaction(cpp11::external_pointer<DbConnectionPtr> con_){
+  DbConnectionPtr* con = con_.get();
+  return con->get()->in_transaction();
```
Comment (krlmlr) [inline src/connection.cpp:73]: Explicit is slightly better here:

```suggestion
  return (con->get()->in_transaction() != 0);
```

**Review by krlmlr — APPROVED** [review]
Great, thanks!

**Comment by krlmlr** [comment]
Thanks!

**Comment by krlmlr** [comment]
@bpvgoncalves: 2d4ac14db3406cbb4eab7ff66acffdb2c3622f0a introduces a problem with the orderly package, found through `git bisect`. That package fails `R CMD check` after this commit. 

Other packages fail too with the current main branch. Can you please take a look?

My `test.sh` for `git bisect` :

```sh
#!/bin/sh

set -ex

git clean -fdx src
R CMD INSTALL .
R -q -e 'rcmdcheck::rcmdcheck("~/git/R/orderly")' 
```

**Comment by bpvgoncalves** [comment]
Hello. I missed this mail last night. I'm really sorry this change broke
something... I'll take a look at it. Best regards.

**Comment by bpvgoncalves** [comment]
So... It looks orderly, and possibly other packages, are at some point applying a commit/rollback when no open transactions exists.
The new functions did not allow this behavior by raising an error.

After applying PR #464 it seem to be fixed.
When testing 'orderly' locally I still receive some errors when R CMD check'ing due failure in latex->pdf conversion for docs. devtools::check(document = FALSE) completes successfully.

**Pull request merged**